linkerd/announcer/curatorsd/src/main/scala/com/medallia/l5d/curatorsd/announcer/CuratorSDAnnouncer.scala

```diff
@@ -13,34 +13,39 @@ import org.apache.curator.x.discovery._
 /**
  * Announcer that uses the curator service discovery format.
  * <p>
- * Format used for the service name Path: /#/com.medallia.curatorsd/{full service name}[#{tenant}]
- * tenant is optional. If it's empty or the last part is missing, it will be registered as a multi-tenant service.
+ * Format used for the service name Path:
+ * /#/com.medallia.curatorsd/protocol/[tenant]/service_name
+ *
+ * tenant is optional, "_" represents multitenant services
  */
 class CuratorSDAnnouncer(zkConnectStr: String) extends FutureAnnouncer {
 
   override val scheme: String = "zk-curator"
 
   private val log = Logger(getClass)
 
-  val serviceDiscoveryInfo = CuratorSDCommon.createServiceDiscovery(zkConnectStr)
+  val serviceDiscoveryInfo:ServiceDiscoveryInfo = CuratorSDCommon.createServiceDiscovery(zkConnectStr)
 
-  private def announce(serviceId: String, tenant: Option[String], address: InetSocketAddress): Future[Announcement] = {
+  private def announce(protocol: String, serviceId: String, tenant: Option[String], address: InetSocketAddress): Future[Announcement] = {
     val tenantStr = tenant.getOrElse("(multi-tenant)")
-    log.info("Announcing %s, tenant: %s address: %s, ZK cluster: %s", serviceId, tenantStr, address, zkConnectStr)
+    log.info("Announcing %s, protocol: %s, tenant: %s, address: %s, ZK cluster: %s", serviceId, protocol, tenantStr, address, zkConnectStr)
 
     val serviceFullPath = CuratorSDCommon.getServiceFullPath(serviceId, tenant)
-
-    // TODO (future) how to specify https? Handle this when we work on the Namer.
     val addressHostString = address.getHostString
     val addressPort = address.getPort
     val builder = ServiceInstance.builder[ServiceInstanceInfo]
       .name(serviceFullPath)
-      .uriSpec(new UriSpec(s"http://$addressHostString:$addressPort"))
-      .port(addressPort)
+      .uriSpec(new UriSpec(s"$protocol://$addressHostString:$addressPort"))
       .address(addressHostString)
       .payload(ServiceInstanceInfo(s"serviceId: $serviceId, tenant: $tenantStr"))
       .serviceType(ServiceType.DYNAMIC)
 
+    if (protocol == "https") {
```

Comment:
Do we want to make the protocol checking return an error in case it is neither http nor https.